src/Donations/Micros/Authentication.tsx

```diff
@@ -43,16 +44,33 @@ function Authentication({}: Props): ReactElement {
   const [openVerifyEmailModal, setopenVerifyEmailModal] = React.useState(false);
 
   const loadUserProfile = async () => {
-    if ((user && user.email_verified) || queryToken) {
+    let token;
+    if (queryToken) {
+      const jwtToken = parseJwt(queryToken);
+      const isTokenValid =
+        new Date(jwtToken.exp * 1000).getTime() > new Date().getTime();
+
+      if (isTokenValid) {
+        token = queryToken;
+      }
+      let queryParams = { ...router.query };
+      delete queryParams.token;
+      router.replace({ query: queryParams });
+    } else if (user && user.email_verified) {
+      token = await getAccessTokenSilently();
+    }
+
+    if ((user && user.email_verified) || token) {
       try {
         // if we have access token in the query params we use it instead of using the
-        const token = queryToken ? queryToken : await getAccessTokenSilently();
```

Comment:
The comment above this delete line does not make sense any more now and should be moved into the new code!